test/e2e/node/kubelet.go

```diff
@@ -479,62 +485,266 @@ var _ = SIGDescribe("kubelet", func() {
+			cmd := tk.KubectlCmd("alpha", "node-logs", "--role", "master", "--query", "kubelet", "-o", "short", "--until-time", start.Format(RFC3339))
+			result := runKubectlCommand(cmd, "--role --query --until-time")
+			logs := journalctlCommand("-u", "kubelet", "-U", "now", "")
+			if result != logs {
+				framework.Failf("Failed to receive the correct kubelet logs")
+			}
+		})
+
+		/*
+			Test if kubectl node-logs --role master --query kubelet -o short
+			returns the true kubelet logs!
+		*/
+
+		ginkgo.It("should return the correct logs for the requested query", func() {
+			ginkgo.By("Starting the command")
+			tk := e2ekubectl.NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, ns)
+			cmd := tk.KubectlCmd("alpha", "node-logs", "--role", "master", "--query", "kubelet", "-o", "short")
+			result := runKubectlCommand(cmd, "--query -o")
+			logs := journalctlCommand("-u", "kubelet", "-b", "0", "")
+			if result != logs {
+				framework.Failf("Failed to receive the correct kubelet logs")
+			}
+		})
+
+		/*
+			Test if kubectl node-logs works for other timezones
+			except UTC!
+		*/
+
+		ginkgo.It("should return the correct logs for the requested query in the correct timezone format", func() {
+			ginkgo.By("Starting the command")
+			tzone := os.Getenv("TZ")
+			os.Setenv("TZ", "Africa/Cairo")
+			tk := e2ekubectl.NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, ns)
+			cmd := tk.KubectlCmd("alpha", "node-logs", "--role", "master", "--query", "kubelet", "-o", "short")
+			result := runKubectlCommand(cmd, "--query -o")
+			logs := journalctlCommand("-u", "kubelet", "-b", "0")
+			if result != logs {
+				framework.Failf("Failed to receive the correct kubelet logs for the specific timezone")
+			}
+			os.Setenv("TZ", tzone)
+		})
 	})
 })
 
-func runKubectlCommand(cmd *exec.Cmd, arg string) {
+func runKubectlCommand(cmd *exec.Cmd, arg string) (result string) {
 	stdout, stderr, err := framework.StartCmdAndStreamOutput(cmd)
+	var buf bytes.Buffer
 	if err != nil {
 		framework.Failf("Failed to start kubectl command: %v", err)
 	}
 	defer stdout.Close()
 	defer stderr.Close()
 	defer framework.TryKill(cmd)
-	buf := make([]byte, 128)
-	if _, err = stdout.Read(buf); err != nil {
+
+	b_read, err := io.Copy(&buf, stdout)
+	if err != nil {
 		framework.Failf("Expected output from kubectl alpha node-logs %s: %v", arg, err)
 	}
-	framework.Logf("output: %s", buf)
+	out := ""
+	if b_read >= 0 {
+		out = buf.String()
+	}
+
+	framework.Logf("output: %s", out)
+	return out
+}
+
+func lookForString(expectedString string, result string) {
+	if strings.Contains(result, expectedString) {
+		return
+	}
+	framework.Failf("Failed to find \"%s\", last result: \"%s\"", expectedString, result)
```

Comment:
what does "last result" means in this context? Did you mean:

```suggestion
	framework.Failf("Failed to find \"%s\" in \"%s\"", expectedString, result)
```